src/main/java/org/spongepowered/common/registry/type/item/ItemStackComparatorRegistryModule.java

```diff
@@ -0,0 +1,115 @@
+
+import com.google.common.collect.Lists;
+import org.spongepowered.api.data.Property;
+import org.spongepowered.api.data.manipulator.DataManipulator;
+import org.spongepowered.api.data.manipulator.mutable.entity.DamageableData;
+import org.spongepowered.api.item.inventory.ItemStack;
+import org.spongepowered.api.item.inventory.ItemStackComparators;
+import org.spongepowered.api.registry.RegistryModule;
+import org.spongepowered.api.registry.util.RegisterCatalog;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+@SuppressWarnings("unchecked")
+public final class ItemStackComparatorRegistryModule implements RegistryModule {
+
+    @RegisterCatalog(ItemStackComparators.class)
+    private final Map<String, Comparator<ItemStack>> comparators = new HashMap<>();
+
+    @Override
+    public void registerDefaults() {
+
+        Comparator<ItemStack> type = Comparator.comparing(i -> i.getType().getId());
+        this.comparators.put("TYPE", type);
+        Comparator<ItemStack> size = Comparator.comparing(ItemStack::getQuantity);
+        this.comparators.put("SIZE", size);
+        Comparator<ItemStack> typeSize = type.thenComparing(size);
+        this.comparators.put("TYPE_SIZE", typeSize);
+        this.comparators.put("DEFAULT", typeSize);
+        Properties properties = new Properties();
+        this.comparators.put("PROPERTIES", properties);
+        ItemDataComparator itemData = new ItemDataComparator();
+        this.comparators.put("ITEM_DATA", itemData);
+        this.comparators.put("ITEM_DATA_IGNORE_DAMAGE", new ItemDataComparator(DamageableData.class));
+        this.comparators.put("IGNORE_SIZE", type.thenComparing(properties).thenComparing(itemData));
+        this.comparators.put("ALL", type.thenComparing(size).thenComparing(properties).thenComparing(itemData));
+    }
+
+    private static final class Properties implements Comparator<ItemStack> {
+
+        @Override
+        public int compare(ItemStack o1, ItemStack o2) {
+            List<Property<?, ?>> properties = Lists.newArrayList(o2.getApplicableProperties());
+            for (Property<?, ?> property : o1.getApplicableProperties()) {
+                if (properties.contains(property)) {
+                    properties.remove(property);
+                } else {
+                    return -1;
+                }
+            }
+            return properties.size();
+        }
+    }
+
+    private static final class ItemDataComparator implements Comparator<ItemStack> {
+
+        private final Class<? extends DataManipulator<?, ?>>[] ignored;
+
+        ItemDataComparator(Class<? extends DataManipulator<?, ?>>... ignored) {
+            this.ignored = ignored;
+        }
+
+        @Override
+        public int compare(ItemStack o1, ItemStack o2) {
+            List<DataManipulator<?, ?>> manipulators = Lists.newArrayList(o2.getContainers());
```

Comment:
Same for here. You might use a `LinkedHashSet` since you need to check everything in a set conditionally.